src/EventProducerService.h

```diff
@@ -33,13 +33,19 @@ class EventProducerService : public Service {
   void sendEvent(bool state, byte evValue, byte data1);
   void sendEvent(bool state, byte evValue, byte data1, byte data2);
   void sendEvent(bool state, byte evValue, byte data1, byte data2, byte data3);
+  void generateMessage();
+  void generateMessage(byte data1);
+  void generateMessage(byte data1, byte data2);
+  void generateMessage(byte data1, byte data2, byte data3);
```

Comment:
These generateMessage() functions should be private. You don't want anyone call these when they might not understand that they need to set opcode and nn_en first.

Reply:
OK